### ISSUE TITLE
iOSConnectNetwork connects network but gets into fail callback - iOS 12+

### ISSUE THREAD

**Issue by gdeiorio** [tripflex/WifiWizard2#62]
﻿

### Prerequisites

Check all boxes if you have done the following:
-  Checked that your issue isn't already filed: https://github.com/tripflex/wifiwizard2/issues
-  Make sure you fill out the Issue Type below

### Issue type
Select all that apply
-  Bug

### Description

When connecting a network, open or protected, with a DISTRIBUTION PROFILE on iOS 12+, it connects correctly, but then it gets into the fail callback and a native dialog of connection failure is shown, although the net stays connected. This doesn't happen with a development profile.

### Steps to Reproduce

1. Pass to function the params of an open or protected network
2. Build with a Distribution profile
3. Connect network

**Expected behavior:** Connect network and go on OR get a failure dialog and don't connect

**Actual behavior:** The network is connected and a failure dialog is shown

**Reproduces how often:** 100% of times

### Versions

WifiWizard 3.1.0;
iPhone 6s;
iOS 12+;
Distribution profile

**Comment by dpinter** [comment]
I have the same problem with WifiWizard2.getConnectedSSID().
It works in development mode, but when I build it with Enterprise certificate, I get a failure callback with NOT AVAILABLE error. Can you help me?

Cordova > 8, iOS > 12, tested on Ipad

**Comment by tripflex** [comment]
@gdeiorio @dpinter unfortunately i don't have any ios devices to test with, have you tried this:

> iOS 12 and later, enable the Access WiFi Information capability for your app in Xcode. When you enable this capability, Xcode automatically adds the Access WiFi Information entitlement to your entitlements file and App ID.

**Comment by luisAzcuaga** [comment]
Hey guys, I solved this issue stated here https://github.com/tripflex/WifiWizard2/issues/60; the fix is to **enable** Access Wifi Information under Capabilities.

**Comment by gdeiorio** [comment]
I had already enabled the Access Wifi Information capability, but the trick worked only in development environment.

Anyway, I solved the issue when I noticed that two different capabilities files were created for development and production: while the development one was created with all the correct capabilities, the production one lacked of almost all of them, so I got to add them manually.

**Comment by tripflex** [comment]
@luisAzcuaga @gdeiorio thanks for your comments, now that i have an iphone i can start testing with, I can start working on checking if maybe this is a specific issue related to Cordova not setting up the dev & production capability files

**Comment by kelter-antunes** [comment]
Managed to get it working by adding:
`mode="add"`
to all the `<config-file` entries.

**Comment by sarpherviz** [comment]
IOS12+ Distribution Certificate Not Available problem solved.

When i run the app in development certificate or adhoc certificate the application is working but when i sign the app with "Distribution Certificate" getConnectedSSID and getConnectedBSSID methods return "Not Available" error.

After this problem platforms->ios-> "ProjectName"->Entitlements-Release.plist and Entitlements-Debug.plist files are different. I fix Entitlements-Release.plist manually and test again , i got same error.

After that i search Entitlements setting in XCODE.

[SOLUTION]: In XCode, at Build Settings tab -> Code Signing Entitlements-> Change Release section to Entitlements-Debug.plist

Build and archive project.
![screen shot 2019-03-05 at 10 26 57](https://user-images.githubusercontent.com/501935/53788298-7677fb00-3f32-11e9-82ee-8b60b94d4fe0.png)

**Comment by tripflex** [comment]
@kelter-antunes thanks for reporting this .. i don't see any configuration for that in the Cordova docs, i'm wondering if we should be using `edit-config` instead of `config-file` 
https://cordova.apache.org/docs/en/latest/plugin_ref/spec.html#edit-config

Will need to do a little bit more research into this

**Comment by tripflex** [comment]
I do see some issue on `cordova-ios` regarding the new build system, which may have some affect on this if using older version of `cordova-ios` prior to 5.0.0:
https://github.com/apache/cordova-ios/issues/407

I guess we could try setting min version of `cordova-ios` to >= 5.0.0 but not sure if that would have any negative affects

**Comment by awaism551** [comment]
Make sure Entitlements-Debug.plist and Entitlements-Release.plist files have network capabilities like I had to manually add this in release file.
`<plist version="1.0">
    <dict>
	<key>com.apple.developer.networking.networkextension</key>
	<array/>
	<key>com.apple.developer.networking.wifi-info</key>
	<true/>
</dict>
</plist>`

**Comment by tripflex** [comment]
@awaism551 yeah just wondering if this is specific to a version of cordova or something else because in my tests this is my generated `Entitlements-Release.plist` and `Entitlements-Debug.plist`:

```
<plist version="1.0">
  <dict>
    <key>com.apple.developer.networking.HotspotConfiguration</key>
    <true/>
    <key>com.apple.developer.networking.networkextension</key>
    <array/>
    <key>com.apple.developer.networking.wifi-info</key>
    <true/>
  </dict>
</plist>
```

**Comment by tripflex** [comment]
@gdeiorio @dpinter @awaism551 @kelter-antunes @luisAzcuaga @sarpherviz 
Can you guys please run `cordova platform version ios` and report back the result.

Here's mine:
Installed platforms:
  android 6.2.3
  ios 4.5.1

**Comment by awaism551** [comment]
Thats my versions:
Installed platforms:
  android 6.2.3
  ios 4.3.1
Actually after I updated my xcode to 10.1 from 9.2, I started having issue with this plugin. For debug, I had to go to capabilities section and turn ON Access wifi info and Network Extensions. This updated my Entitlements-Debug.plist file but Entitlements-release.plist was not updating though I also made a release build with xcode. After manually adding in last, I got success

**Comment by sarpherviz** [comment]
android 7.1.4
ios 4.5.5

**Comment by dpinter** [comment]
android 7.1.4
ios 4.5.5

**Comment by 13546777510** [comment]
> IOS12 +分发证书不可用问题解决了。
> 
> 当我在开发证书或adhoc证书中运行应用程序时，应用程序正在运行，但是当我使用“分发证书”签署应用程序时，getConnectedSSID和getConnectedBSSID方法返回“Not Available”错误。
> 
> 在此问题平台 - > ios->“ProjectName” - > Entitlements-Release.plist和Entitlements-Debug.plist文件不同之后。我手动修复了Entitlements-Release.plist并再次测试，我得到了同样的错误。
> 
> 之后，我在XCODE中搜索Entitlements设置。
> 
> [解决方案]：在XCode中，在Build Settings选项卡 - > Code Signing Entitlements-> Change Release部分到Entitlements-Debug.plist
> 
> 构建和归档项目。
> ![屏幕截图2019-03-05 at 10 26 57](https://user-images.githubusercontent.com/501935/53788298-7677fb00-3f32-11e9-82ee-8b60b94d4fe0.png)

@sarpherviz But why do I reproduce this error in a few days, I have to check this error every time I package it, and I have to use this method to fix it. Is there any permanent solution?
